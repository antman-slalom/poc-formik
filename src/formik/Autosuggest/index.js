import Component from './Autosuggest';

export default Component;
