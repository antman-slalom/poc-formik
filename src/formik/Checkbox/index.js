import Component from './Checkbox';

export default Component;
