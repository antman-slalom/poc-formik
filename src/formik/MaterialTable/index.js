import Component from './MaterialTable';

export default Component;
