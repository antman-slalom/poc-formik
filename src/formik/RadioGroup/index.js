import Component from './RadioGroup';

export default Component;
