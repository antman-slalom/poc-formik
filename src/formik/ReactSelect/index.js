import Component from './ReactSelect';

export default Component;
