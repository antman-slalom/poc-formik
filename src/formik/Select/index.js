import Component from './Select';

export default Component;
