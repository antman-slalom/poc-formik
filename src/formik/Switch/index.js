import Component from './Switch';

export default Component;
