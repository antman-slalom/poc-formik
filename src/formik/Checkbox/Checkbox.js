import React from 'react';

import { fieldToCheckbox } from 'formik-material-ui';
import isEqual from 'react-fast-compare';
import { Field } from 'formik';
import MuiCheckbox from '@material-ui/core/Checkbox';

import FormHelperText from '@material-ui/core/FormHelperText';
import FormControl from '@material-ui/core/FormControl';
import FormControlLabel from '@material-ui/core/FormControlLabel';

import _get from 'lodash/get';

class CheckboxWithLabel extends React.Component {
    shouldComponentUpdate(nextProps) {
        return !isEqual(nextProps, this.props);
    }

    render() {
        const {
            label,
            name,
            onChange,
            onBlur,
            ...others
        } = this.props;

        return (
            <Field name={ name } >
                {
                    ({ field, form, meta }) => {
                        const params = {
                            ...fieldToCheckbox({
                                field,
                                form,
                                disabled: others.disabled,
                                ...others
                            })
                        };

                        const error = _get(form.errors, name);
                        const errorText = _get(form.touched, name) ? error : '';

                        return (
                            <FormControl error={ !!errorText } label={ label } variant={ others.variant }>
                                <FormControlLabel control={
                                    <MuiCheckbox
                                        { ...params }

                                        checked= { Boolean(field.value) }

                                        onBlur={ (event) => {
                                            field.onBlur(event);

                                            if (typeof onBlur === 'function') {
                                                onBlur(event);
                                            }
                                        } }

                                        onChange={ (event) => {
                                            field.onChange(event);

                                            if (typeof onChange === 'function') {
                                                onChange(event);
                                            }
                                        } }

                                        value={ field.value ? 'checked' : '' }
                                    />
                                }
                                label={ label }
                                />

                                <FormHelperText>{ errorText }</FormHelperText>
                            </FormControl>
                        );
                    }
                }
            </Field>
        );
    }
}

export default CheckboxWithLabel;
