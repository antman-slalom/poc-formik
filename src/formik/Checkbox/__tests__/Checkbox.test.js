import React from 'react';
import { shallow } from 'enzyme';
import { Field } from 'formik';
import MuiCheckbox from '@material-ui/core/Checkbox';
import FormControlLabel from '@material-ui/core/FormControlLabel';

import FormControl from '@material-ui/core/FormControl';
import Checkbox from '../Checkbox';

describe('Checkbox', () => {
    let wrapper;
    let instance;

    const props = {
        name: 'checkbox-test'
    };

    beforeEach(() => {
        wrapper = shallow(<Checkbox { ...props } />);
        instance = wrapper.instance();
    });

    afterEach(() => {
        // clears all mocks to prevent mock bleeding from test to test,
        jest.clearAllMocks();

        // mock the API service after the clear
        jest.mock('../../../services/AdminService');
    });

    // =================================
    // Render and Snapshot
    // =================================

    it('matches snapshot', () => {
        expect(wrapper.instance()).toMatchSnapshot();
    });

    it('should render', () => {
        expect(wrapper.instance()).not.toBeNull();
    });

    // =================================
    // Components call correct functions
    // =================================

    // TODO - prove this part works in test... Will help!!
    it('should call DTM onClick function', () => {
        const onBlur = jest.fn();

        const wrapperOnBlur = shallow(<Checkbox onBlue={ onBlur } { ...props } />).dive();

        expect(onBlur).toHaveBeenCalledTimes(0);
        wrapperOnBlur.find('#checkbox-test').simulate('blur', null);
        expect(onBlur).toHaveBeenCalledTimes(1);
    });

    // =================================
    // Functions act as expected
    // =================================
    it('Test shouldComponentUpdate to be false', () => {
        const result = instance.shouldComponentUpdate(props, wrapper.state());
        expect(result).toBe(false);
    });

    it('Test shouldComponentUpdate to be true', () => {
        const result = instance.shouldComponentUpdate({}, {});
        expect(result).toBe(true);
    });
});
