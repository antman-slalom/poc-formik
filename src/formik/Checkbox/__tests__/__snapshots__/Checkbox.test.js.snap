// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox matches snapshot 1`] = `
CheckboxWithLabel {
  "context": Object {},
  "props": Object {
    "name": "checkbox-test",
  },
  "refs": Object {},
  "setState": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useResponder": [Function],
        "useState": [Function],
      },
      "_element": <CheckboxWithLabel
        name="checkbox-test"
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <FormikConnect(FieldInner)
        name="checkbox-test"
      >
        [Function]
      </FormikConnect(FieldInner)>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;
